roxie/ccd/ccdmain.cpp

```diff
@@ -504,11 +630,36 @@ int STARTQUERY_API start_query(int argc, const char *argv[])
     {
         Owned<IProperties> globals = createProperties(true);
         for (int i = 1; i < argc; i++)
+        {
+            std::string name, value;
+            if (splitarg(argv[i], name, value))
+            {
+                if (name=="--topologyServer")
+                {
+                    topologyServers.append(SocketEndpoint(value.c_str()));
+                    continue;
+                }
+                else if (name=="--serverPort")
+                {
+                    farmerPorts.push_back(atoi(value.c_str()));
+                    continue;
+                }
+                else if (name=="--channel")
+                {
+                    slaveChannels.push_back(atoi(value.c_str()));
+                    continue;
+                }
+            }
             globals->loadProp(argv[i], true);
-
+        }
         Owned<IFile> sentinelFile = createSentinelTarget();
         removeSentinelFile(sentinelFile);
 
+        if (globals->hasProp("--topologyServer"))
+        {
+            topologyServers.append(SocketEndpoint(globals->queryProp("--topologyServer")));
```

Comment:
minor: is this now redundant because of the code above?

Reply:
Yes